front/migrations/20240916_backfill_ds_in_agent_table_query_configurations.ts

```diff
@@ -0,0 +1,106 @@
+import type { LightWorkspaceType } from "@dust-tt/types";
+import assert from "assert";
+import { Op } from "sequelize";
+
+import { Authenticator } from "@app/lib/auth";
+import { AgentTablesQueryConfigurationTable } from "@app/lib/models/assistant/actions/tables_query";
+import { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import { DataSourceViewResource } from "@app/lib/resources/data_source_view_resource";
+import { VaultResource } from "@app/lib/resources/vault_resource";
+import type { Logger } from "@app/logger/logger";
+import { makeScript, runOnAllWorkspaces } from "@app/scripts/helpers";
+
+async function backfillDataSourceIdInAgentTableQueryConfigurationForWorkspace(
+  workspace: LightWorkspaceType,
+  logger: Logger,
+  execute: boolean
+) {
+  const auth = await Authenticator.internalAdminForWorkspace(workspace.sId);
+
+  // List workspace data sources.
+  const dataSources = await DataSourceResource.listByWorkspace(auth);
+
+  logger.info(
+    `Found ${dataSources.length} data sources for workspace(${workspace.sId}).`
+  );
+
+  const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
+
+  // Retrieve data source views for data sources.
+  const dataSourceViews =
+    await DataSourceViewResource.listForDataSourcesInVault(
+      auth,
+      dataSources,
+      globalVault
+    );
+
+  // Count agent tables query configurations that uses those data sources and have no dataSourceIdNew.
+  const agentTablesQueryConfigurationsCount =
+    await AgentTablesQueryConfigurationTable.count({
+      where: {
+        // /!\ `dataSourceId` is the data source's name, not the id.
+        dataSourceId: dataSources.map((ds) => ds.name),
+        dataSourceIdNew: {
+          [Op.is]: null,
+        },
+        // Given that the name isn't unique we need to count per workspace.
+        dataSourceWorkspaceId: workspace.sId,
+      },
+    });
+
+  logger.info(
+    `About to update ${agentTablesQueryConfigurationsCount} agent tables query configurations for workspace(${workspace.sId}).`
+  );
+
+  if (!execute) {
+    return;
+  }
+
+  for (const ds of dataSources) {
+    const dataSourceView = dataSourceViews.find(
+      (dsv) => dsv.dataSourceId === ds.id
```

Comment:
We have no guarantee to have just one even if that should be the case

Reply:
Since our search uses the global workspace vault, even though this guarantee isn't explicitly represented in the code, the expectations are that we should have maximum one view per data source per vault.

Let me slightly update this code to error if we have more than one.